### PR TITLE
Upgrade ds to v48

### PR DESCRIPTION
**Pull request by rmccar**
### What is the context of this PR?
This PR upgrades the DS to v48 which will fix the validation error message spacing bug and the header links disappearing on smaller screens.

Also fixes breaking change adding `toggleServicesButton` to service links

### How to review 
- Tests pass
- Check that duration validation error boxes are now spaced correctly (`test_date_range` schema enter a date in the first field and submit)
- Test any schema and when screen size is reduced see that the sign out and account links go into a menu button

### Checklist

* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

**Comment by rmccar** [comment]
[Validation error message bug](https://trello.com/c/DynEZAGK/5235-validation-error-message-bug)

**Review by MebinAbraham — CHANGES_REQUESTED** [review]
The new Menu dropdown is classed as a button in the source. If you try to click it, it will just submit the page.

File: tests/app/helpers/test_template_helpers.py [hunk 1]
```diff
@@ -160,6 +160,10 @@ def test_get_page_header_context_census_nisra(app: Flask):
             BusinessSurveyConfig(),
             True,
             {
+                "toggleServicesButton": {
+                    "text": "Menu",
```
Comment (MebinAbraham) [inline tests/app/helpers/test_template_helpers.py:164]: This should be marked for translation.

File: app/helpers/template_helpers.py [hunk 2]
```diff
@@ -73,12 +73,20 @@ def context(self) -> dict[str, Any]:
...
+        self,
+    ) -> Optional[dict[str, Union[dict[str, str], list[dict]]]]:
         if service_links := self._survey_config.get_service_links(
             sign_out_url=self._sign_out_url,
             is_authenticated=current_user.is_authenticated,
         ):
-            return {"itemsList": service_links}
+            return {
+                "toggleServicesButton": {
+                    "text": "Menu",
```
Comment (MebinAbraham) [inline app/helpers/template_helpers.py:85]: This should be marked for translation extraction using `_()`.

**Comment by rmccar** [comment]
> The new Menu dropdown is classed as a button in the source. If you try to click it, it will just submit the page.

Fixed in new DS PR (https://github.com/ONSdigital/design-system/pull/2133)

File: app/helpers/template_helpers.py [hunk 3]
```diff
@@ -73,12 +73,20 @@ def context(self) -> dict[str, Any]:
...
+    ) -> Optional[dict[str, Union[dict[str, str], list[dict]]]]:
         if service_links := self._survey_config.get_service_links(
             sign_out_url=self._sign_out_url,
             is_authenticated=current_user.is_authenticated,
         ):
-            return {"itemsList": service_links}
+            return {
+                "toggleServicesButton": {
+                    "text": lazy_gettext("Menu"),
+                    "ariaLabel": lazy_gettext("Toggle services menu"),
```
Comment (MebinAbraham) [inline app/helpers/template_helpers.py:86]: Non blocker: Historically, we have avoided marking aria labels for translations. Prob needs another revisit as we have a mix now, I think mostly unmarked. I think it was driven by whether screen readers etc support non-eng.

Comment (rmccar) [reply app/helpers/template_helpers.py]: Will remove the aria label stuff for now and add a card to look into screenreader language support

**Review by MebinAbraham — APPROVED** [review]

**Review by petechd — APPROVED** [review]
Validation error spacing correct and menu list as expected.

**Pull request merged**